src/api/tenants/index.ts

```diff
@@ -0,0 +1,294 @@
+/* Copyright (c) 2022, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+import { getApiUrl, useFetchData } from "../../utils";
+import { ProviderConfig, ProviderConfigResponse, TenantInfo } from "./types";
+
+export const useTenantCreateService = () => {
+	const fetchData = useFetchData();
+
+	const createOrUpdateTenant = async (
+		tenantId: string
+	): Promise<
+		| {
+				status: "OK";
+				createdNew: boolean;
+		  }
+		| {
+				status: "MULTITENANCY_NOT_ENABLED_IN_CORE_ERROR";
+		  }
+		| {
+				status: "INVALID_TENANT_ID_ERROR";
+				message: string;
+		  }
+		| undefined
+	> => {
+		const response = await fetchData({
+			url: getApiUrl("/api/tenant"),
+			method: "PUT",
+			config: {
+				body: JSON.stringify({
+					tenantId,
+				}),
+			},
+		});
+
+		if (response.ok) {
+			const body = await response.json();
+			return body;
+		}
+
+		return undefined;
```

Comment:
throw an error in this case, and handle it correctly

Reply:
Removed undefined from all APIs throwing errors everywhere now